docs/generate-time-series/javascript.rst

```diff
@@ -0,0 +1,232 @@
+last 1, 5 and 15 minutes.
+
+Set up CrateDB
+==============
+
+First, import the `pg`_ client::
+
+    > const { Client } = require('pg')
+
+Then, `connect`_ to CrateDB, using the port for the `Postgres Wire Protocol`_::
+
+    > const client = new Client({
+    ...   user: 'crate',
+    ...   host: 'localhost',
+    ...   database: 'doc',
+    ...   port: 5432,
+    ... })
+    > await client.connect()
+
+The client can also be initialised with a single connection string, if prefered::
+
+    > const client = new Client({
+    ...   connectionString: 'postgresql://crate@localhost:5432/doc'
+    ... })
+    > await client.connect()
+
+Then, finally, `create a table`_ suitable for writing load averages::
+
+    > var query = `
+    ...     CREATE TABLE load (
+    ...         timestamp TIMESTAMP GENERATED ALWAYS AS CURRENT_TIMESTAMP, 
+    ...         avg_1m REAL, 
+    ...         avg_5m REAL, 
+    ...         avg_15m REAL)`
+    > await client.query(query)
+
+
+In the `CrateDB Admin UI`_, you should see the new table when you navigate to
+the *Tables* screen using the left-hand navigation menu:
+
+.. image:: ../_assets/img/generate-time-series/table.png
+
+
+Record your system load
+=======================
+
+With the table in place, you can start recording load averages.
+
+The following command calls `os.loadavg`_ and uses the result triple as `input
+values`_ for the `INSERT`_ query::
+
+    > await client.query("INSERT INTO load (avg_1m, avg_5m, avg_15m) VALUES (?, ?, ?)", os.loadavg())
+
+Press the up arrow on your keyboard and hit *Enter* to run the same command a
+few more times.
+
+When you're done, you can `SELECT`_ that data back out of CrateDB, like so::
+
+    > client.query('SELECT * FROM load ORDER BY timestamp DESC', (err, res) => {
+    ...   if (err) {
+    ...       console.log(err.stack)
+    ...     } else {
+    ...       console.log(res.rows)
+    ...     }
+    ... })
+    [ { timestamp: 2020-02-13T14:23:23.566Z,
+        avg_1m: 0.83251953,
+        avg_5m: 0.54785156,
+        avg_15m: 0.2265625 },
+      { timestamp: 2020-02-13T14:23:22.968Z,
+        avg_1m: 0.90527344,
+       avg_5m: 0.5571289,
+        avg_15m: 0.22802734 },
+      { timestamp: 2020-02-13T14:23:21.144Z,
+        avg_1m: 0.90527344,
+        avg_5m: 0.5571289,
+        avg_15m: 0.22802734 } ]
+
+Here we have recorded three sets of load averages with a corresponding
+timestamp. We can now close the connection::
+
+    > await client.end()
+
+
+Automate it
+===========
+
+Now we have the basics figured out, let's automate the data collection.
+
+Our approach will change a little bit here. We were previously using a `Client`_ to
+connect to and insert data into CrateDB. However, Clients are ephemeral, and once closed
+they need to be recreated. However, creating a new Client requires a handshake with CrateDB, and
+this overhead cost can be prohibative if we are creating new clients rapidly.
```

Comment:
I'm curious

if you look at the python tutorial, you will notice that I create a new connection each time the main loop evals. I did this primarily to mimic the behavior of the shell script tutorial

but then it occurred to me that keeping open a long-running connection invites problems. if a network issue arises, that while loop is going to start producing errors

I also didn't want to complicate the script too much by introducing connection management

so I would ask, which of the following sounds better to you:

1. drop the connection pool stuff for node. and just try to recreate the connection each time, mimicking the existing two tutorials (and building in a little resiliency -- though not much, because we're still not doing exception handling)

2. same as (1) but add exception handling 

3. keeping the connection pool use here, and updating the python tutorial to do something similar

4. keeping the connection pool use here, and not updating the python tutorial

Reply:
I would probably say 3. The crate-python driver already does pooling under the hood, see: https://github.com/crate/crate-python/blob/750fe0042c25b8c7a77ffb8b3afa096926f1fd8d/src/crate/client/http.py#L92 - when you do `connect` with crate-python, it gets a connection pool from urllib3 and manages the connection with that. With the postgres driver, we have to do that explicitly. In the python driver, a connection is actually a wrapper around a connection pool. It's probably not needed to create a new connection on every main loop of the python driver, actually.

Reply:
cool okay. let's do that then

after we merge this pr, do you wanna submit a patch for the python tutorial as you have more familiarity with the client? (otherwise, happy to look into it myself!)

Reply:
Will do!